unity-client/Assets/Plugins/RequestAnimationFrameBackground/RequestAnimationFrameBackground.jspre

```diff
@@ -1,41 +0,0 @@
-
-(function () {
-  if (window.requestAnimationFrame && document) {
-    window.backgroundFPS = 60;
-    window.__requestAnimationFrame = window.requestAnimationFrame;
-    window.__CURRENT_RAFS = {};
-
-    function backgroundRAF(rafCallback) {
-      setTimeout(function(){
-        rafCallback(performance.now())
-      }, 1000 / window.backgroundFPS);
-    }
-
-    window.requestAnimationFrame = function (rafCallback) {
-      if (document.hidden) {
-        backgroundRAF(rafCallback);
-      } else {
-        var rafId = __requestAnimationFrame(function(stamp) {
-          rafCallback(stamp);
-          delete __CURRENT_RAFS[rafId];
-        });
-        __CURRENT_RAFS[rafId] = rafCallback;
-      }
-    };
-
-    function switchToBackground() {
```

Comment:
@menduz @pravusjif Still, I think we should test it thoroughly. I remember that it was trickier than just calling setTimeout when the document was hidden.

I may have been wrong and overcomplicated it. But if I recall correctly the problem was that if you switched to background in the exact moment a requestAnimationFrame was pending to be executed, it didn't execute until the focus came back to the tab, effectively pausing the tab in background (because it is the RAF callback that ends up calling RAF again).

The solution I came up was this subscription to visibility change, that canceled pending RAFs, and switched them to setTimeout.

I'll try to test if we are having the problem in this branch in a bit.

Reply:
Hmm makes sense.. Upon your confirmation we will make the changes

Reply:
Unfortunately, I can confirm the issue exists.

To reproduce it:
- Enter the explorer and authenticate to start the loading process
- Immediately change tab or minimize the browser (ensure the tab is hidden somehow)
- Wait. Normal loading shouldn't take more than a minute.
- Go back to the tab. You should see that the loading process is still going

You maaaay get lucky and change tabs in the right moment so that the change to setTimeout is successful. But 3/3 times I tested I could reproduce the issue.

If you need help implementing a fix for this let me know :)

Reply:
Cool, thanks

Reply:
Fuck, ok, good catch, tomorrow I'll re-implement that previous fix and add comments to clarify on this so that we don't remove it in the future as well

Reply:
@pablitar can you confirm that with the last change the background running fix is back?

Reply:
seems to work great!